community/governance.adoc

```diff
@@ -0,0 +1,87 @@
+
+[[roles]]
+== Roles and Responsibilities
+
+- **Contributors:** Anyone who contributes to Hibernate projects in any form.
+- **Members:** Contributors eligible for write access to a Hibernate code repository.
+  Responsible for reviewing and merging contributions in their area of expertise.
+- **Project Leaders:** Members (at most one per code repository) with higher decision power whenever the project they lead is affected.
+  Responsible for steering project direction, and for enforcing compliance with requirements of the Commonhaus Foundation.
+
+Small or inactive projects may not have a leader,
+in which case Members interested in the project will steer the project direction.
+
+[[decision-making]]
+== Decision-Making
+
+Hibernate projects follow a common decision-making process.
+
+Consensus-seeking (lazy consensus)::
+Hibernate projects primarily aim for a consensus-based decision-making process,
+where Members and active contributors discuss and come to an agreement.
++
+In practice, this involves:
++
+* Discussing matters openly, to facilitate others joining the discussions and expressing concerns.
+* Taking into account every contributor's opinion, regardless of their role.
+
++
+Actual implementation of consensus decision-making is up to Members and can vary based on the audience and criticality of the discussion.
+Inspiration may be found in
+the https://community.apache.org/committers/decisionMaking.html[Lazy Consensus model as defined by the Apache Foundation],
+and in https://digitalcommons.unl.edu/cgi/viewcontent.cgi?article=1825&context=sociologyfacpub[Martha's Rules].
+Conflict Resolution::
+If conflicts arise, Members are responsible for facilitating a resolution.
++
+Project Leaders hold the power to make the final decision, be it individually for the project they lead,
+or collectively for cross-project matters.
++
+As a last resort, in particular in case of disagreement about the decision-making process,
+the https://www.commonhaus.org/bylaws/cf-council.html[Commonhaus Foundation Council] (CFC) can be asked to mediate the discussion.
+
+[[role-granting-revoking]]
+== Role granting/revoking
+
+The role of Member or Project Leader is granted or revoked through the <<decision-making,decision-making process>>,
+with additional restrictions:
+
+1. The discussion must happen on the Hibernate development mailing list, as listed in the link:/community[Community page on this website].
+// This prevents a Project Leader overruling their own revocation, in particular.
+2. The opinion of the Members or Project Leader whose role is being discussed does not factor into the decision.
+// This is long on purpose, to eliminates the risk of a decision being taken "in absentia" during e.g. holidays.
+// The assumption is that decisions around the project can be taken collectively, or by the previous leader, in the interim.
+3. Discussions regarding the role of Project Leader may not last less than 30 days.
```

Comment:
@emmanuelbernard (and others) due to concerns around the voting system, I removed any mention of votes in order to better reflect the existing process.
I guess we can add voting later if "enlightened dictatorship" proves to be a real dealbreaker for some prospective contributors -- though I'm not sure how we'd ever hear about that.

This is where most changes are, but see the second commit for detailed changes.